test/examples/test_examples.py

```diff
@@ -0,0 +1,166 @@
+    #     pass
+
+    def test_config_unencrypted_example(self):
+        from config_unencrypted_example import ConfigUnencryptedExample
+
+        example = ConfigUnencryptedExample(self.bolt_uri, self.user, self.password)
+
+        self.assertIsInstance(example, ConfigUnencryptedExample)
+
+    # def test_custom_auth_example(self):
+    #     from custom_auth_example import CustomAuthExample
+    #     pass
+
+    def Xtest_cypher_error_example(self):
+        from cypher_error_example import CypherErrorExample
+
+        example = CypherErrorExample(self.bolt_uri, self.user, self.password)
+        employee_number = example.get_employee_number('Alice')
+
+        # FIXME: also check the error output like in the Java example
+        self.assertEqual(employee_number, -1)
+
+    def test_driver_lifecycle_example(self):
+        from driver_lifecycle_example import DriverLifecycleExample
+
+        example = DriverLifecycleExample(self.bolt_uri, self.user, self.password)
+        example.close()
+
+        self.assertIsInstance(example, DriverLifecycleExample)
+
+    def test_hello_world_example(self):
+        import io
+        from contextlib import redirect_stdout
+        from hello_world_example import HelloWorldExample
+
+        f = io.StringIO()
+        with redirect_stdout(f):
+            example = HelloWorldExample(self.bolt_uri, self.user, self.password)
+            example.print_greeting("hello, world")
+            example.close()
+
+        self.assertTrue(f.getvalue().startswith("hello, world, from node"))
+        
+    def test_read_write_transaction_example(self):
+        from read_write_transaction_example import ReadWriteTransactionExample
+
+        example = ReadWriteTransactionExample(self.bolt_uri, self.user, self.password)
+        node_count = example.add_person('Alice')
+
+        self.assertTrue(node_count > 0)
+
+    def test_result_consume_example(self):
+        from result_consume_example import ResultConsumeExample
+
+        self.write("CREATE (a:Person {name: 'Alice'})")
+        self.write("CREATE (a:Person {name: 'Bob'})")
+        example = ResultConsumeExample(self.bolt_uri, self.user, self.password)
+        people = list(example.get_people())
+
+        self.assertEqual(['Alice', 'Bob'], people)
+
+    def test_result_retain_example(self):
+        from result_retain_example import ResultRetainExample
+
+        self.write("CREATE (a:Person {name: 'Alice'})")
+        self.write("CREATE (a:Person {name: 'Bob'})")
+        example = ResultRetainExample(self.bolt_uri, self.user, self.password)
+        example.add_employees('Acme')
+        employee_count = self.read("MATCH (emp:Person)-[:WORKS_FOR]->(com:Company) WHERE com.name = 'Acme' RETURN count(emp)").single()[0]
+
+        self.assertEqual(employee_count, 2)
+
+    # def test_service_unavailable_example(self):
+    #     from service_unavailable_example import ServiceUnavailableExample
+    #     pass
+
```

Comment:
I am not sure how to do the equivalent of `neo4j.stop()` that we do in the Java version.